### DISCUSSION TITLE
Can it have horizontal and downward direction ?

### DISCUSSION THREAD

**Discussion by skylum1** [darioielardi/flutter_speed_dial#114] [Q&A] [answered]
Just like in this demo :
[https://material.angularjs.org/latest/demo/fabSpeedDial](https://material.angularjs.org/latest/demo/fabSpeedDial)

**Comment by prateekmedia** [comment]
We already have direction enums for Up and down, will look into other two when free as they are diffrent from others.

**Comment by skylum1** [comment]
I have tried it but I cannot use the orientation down because fab is always positioned at the bottom right . I have tried this :-
```
Scaffold(
      appBar: AppBar(title: Text('Flutter Speed Dial')),
      body: Stack(children: [
        buildBody(),
        Center(
          child: buildSpeedDial(),
        )
      ]),
      // floatingActionButton: buildSpeedDial(),
    );
```
This is from the example code :- https://pub.dev/packages/flutter_speed_dial/example
I have only done changes in Scaffold Widget.
And I got this behavior . Video link :-[https://filebin.net/v1ixy7yjxpdl9qnk](https://filebin.net/v1ixy7yjxpdl9qnk)

Please tell me how to position fab anywhere I want.

**Comment by prateekmedia** [comment]
> I have tried it but I cannot use the orientation down because fab is always positioned at the bottom right . I have tried this

~~Firstly use the git url to get the plugin as this plugin is not actively updated in pub.dev and secondly~~ currently for some reasons(due to stack and the way this plugin is set up) you cannot position the speed dial.

However you can always change the margins to tweak the position of the speed dial.

**Comment by skylum1** [comment]
Thank you for the response, I was able get it to work with margin property as you suggested.

**Comment by prateekmedia** [comment, marked as answer]
@skylum1 in v4 you can have all the directions you specified
